spec/forms/project_form_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+describe ProjectForm do
+  # describe "validations" do
+
+    # before do
+      # allow_any_instance_of(SlackAdapter).to receive(:find_channel_by_name).and_return(false)
+    # end
+
+    # it { should validate_presence_of :name }
+    # it { should validate_with :channel_doesnt_exist }
+  # end
+
+  describe "#save" do
+    context "when valid" do
+      it "calls persist!" do
+        user = FactoryGirl.create(:user)
+        form = ProjectForm.new(name: "name", description: "description", user: user, slack: slack_double, github: double)
+        expect(form).to receive(:persist!).and_return(true)
+        form.save
+      end
+    end
+
+    context "when invalid" do
+      it "returns false" do
+        form = ProjectForm.new(name: "", description: "description", slack: slack_double)
+        expect(form.save).to eq false
+      end
+    end
+  end
+
+  describe "#persist!" do
+    let(:user) { FactoryGirl.create(:user) }
+    let(:form) { ProjectForm.new(name: "sample project", description: '', user: user, slack: slack_double, github: double) }
```

Comment:
Line is too long. [124/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.